src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -636,28 +640,36 @@ protected void RemoveProjectReference(ProjectReference projectReference)
 
         private static void OnDocumentOpened(object sender, bool isCurrentContext)
         {
-            IVisualStudioHostDocument document = (IVisualStudioHostDocument)sender;
-            AbstractProject project = (AbstractProject)document.Project;
-
-            if (project._pushingChangesToWorkspaceHosts)
-            {
-                project.ProjectTracker.NotifyWorkspaceHosts(host => host.OnDocumentOpened(document.Id, document.GetOpenTextBuffer(), isCurrentContext));
-            }
-            else
+            // We do not want to allow message pumping/reentrancy when processing project system changes.
+            using (Dispatcher.CurrentDispatcher.DisableProcessing())
```

Comment:
Indeed we did fix that after update 2 as part of a0bfff to fix this very same bug.  Wonder why I didn't resolve it then.